docs/proposals/local-tasks.adoc

```diff
@@ -0,0 +1,44 @@
+= Proposal: Local tasks
+
+== Purpose
+
+This proposal aims to adress the following issues:
+
+- Sidecars are barely usable (https://github.com/opendevstack/ods-pipeline/issues/135)
+- Resource Consumption is not configurable from pipeline (https://github.com/opendevstack/ods-pipeline/issues/372)
+
+== Background
+
+Both issues originate from the fact that most things within a Tekton task can only be configured by task authors, and not by pipeline authors. In an ideal world, Tekton would allow sidecars to be defined from a pipeline, as well as allow to configure resource consumption (CPU/memory) of the tasks.
+
+At the time of writing, it does not look like there will be quick solutions to this. One reason for this is that no solution is defined or implemented yet in Tekton, another that uptake in OpenShift would be delayed once there is a solution in Tekton.
+
+The following is a proposal that completely sidesteps waiting for a solution, comes with a few other advantages, and therefore is worth considering in my opinion.
+
+== Solution
+
+ODS pipeline is comprised of two installations: a central one and a project-specific one. The central installation is split into two Helm charts: one for images, and one for (cluster) tasks. The project specific Helm chart contains the Tekton Trigger resources and config maps / secrets.
+
+The proposal is to remove the (cluster) task Helm chart from the central installation, and merge it into the project-specific installation. The tasks would then not be of type `Task` anymore, but of kind `Task`. This might sound like an odd suggestion at first but I think it is an interesting thought.
```

Comment:
```suggestion
The proposal is to remove the (cluster) task Helm chart from the central installation, and merge it into the project-specific installation. The tasks would then not be of type `ClusterTask` anymore, but of kind `Task`. This might sound like an odd suggestion at first but I think it is an interesting thought.
```